bolt/include/bolt/Core/MCPlusBuilder.h

```diff
@@ -54,7 +54,6 @@ enum class IndirectBranchType : char {
   POSSIBLE_TAIL_CALL,      /// Possibly a tail call.
   POSSIBLE_JUMP_TABLE,     /// Possibly a switch/jump table.
   POSSIBLE_PIC_JUMP_TABLE, /// Possibly a jump table for PIC.
-  POSSIBLE_GOTO,           /// Possibly a gcc's computed goto.
```

Comment:
Please don't remove it. I have a WIP diff that adds support and it's known to be present in cpython interpreter loop (_PyEval_EvalFrameDefault,_PyEval_EvalFrame_AOT_Interpreter)

Reply:
If you ask so. But may I ask what is the difference between POSSIBLE_TAIL_CALL, POSSIBLE_GOTO and POSSIBLE_FIXED_BRANCH? I think we need to do something with it, it looks nearly the same to me, at least from ARM perspective.

Reply:
As I understand POSSIBLE_FIXED_BRANCH is signle-entry jump table. Shouldn't we add the same logic to POSSIBLE_TAIL_CALL with createUncondBranch to POSSIBLE_TAIL_CALL @maksfb?

Reply:
My understanding is that tailcall is inter-function control transfer, computed goto is an indirect internal control transfer with a slightly more general address calculation pattern at least on x86 (jump table base address is computed and stored in a register?), and a possible fixed branch is an internal indirect branch to a single target known statically. So it's partially overlapping but pretty well defined classification.

Reply:
Well we're checking tailcall target after we would "classify" it as tailcall, so it might be both local and global. As ffor possible fixed branch - the same, it is even checked to be relaxed only if it is local local, but not on global. As for x86 I imagine you would return this value on some kind of special pattern found, but generally speaking it seems like one of two anyway. I don't really mind, just think it needs to have better description, probably target-dependent, I imagine that on ARM we might use it a bit other way, right now it has some overlap target-specific meaning, but generally seems to overlap, yes